src/ui/BlockEditor.tsx

```diff
@@ -642,12 +649,13 @@ class BlockEditor extends Component<BlockEditorProps> {
     dispatch((_, getState) => {
       const {cur} = getState();
       if (!this.mouseUsed && (cur === null)) {
-        // NOTE(Oak): use setTimeout so that the CM cursor will not blink
-        setTimeout(() => this.props.activateByNid(null, {allowMove: true}), 0);
+        // NOTE(Emmanuel): setAfterDOMUpdate so that the CM cursor will not blink
+        this.pendingTimeout = setAfterDOMUpdate(() => 
+          this.props.activateByNid(null, {allowMove: true}));
         this.mouseUsed = false;
       } else if(this.mouseUsed && (cur === null)) {
         // if it was a click, get the cursor from CM
-        setTimeout(() => this.props.setCursor(ed.getCursor()));
+        this.pendingTimeout = setAfterDOMUpdate(() => this.props.setCursor(ed.getCursor()));
```

Comment:
same problem here with not canceling the previous timeout before creating a new one.